### DISCUSSION TITLE
BelongsToMany relation does not use pivot table's timestamps

### DISCUSSION THREAD

**Discussion by kassah** [laravel/framework#39728] [Q&A] [answered]
- Laravel Version: v8.73.1
- PHP Version: 8.0.12
- Database Driver & Version: mysqlnd 8.0.7

### Description:

When a table's updated_by or created_by columns are customized on a table, and a belongsToMany relation is setup with timestamps, a request via the relation generates an exception as it requests the timestamp columns of the source table, and not of the pivot table.

### Steps To Reproduce:

GitHub repository demonstrating the issue: https://github.com/kassah/belongsToManyTimeStampIssue

Steps:
1. Create a new project
2. Create a source table & model (i.e. Article)
3. Create a pivot table & model (i.e. Review)
4. Create a target table & model (i.e. User) [I re-used the existing User model]
5. Add a relation with belongs to many and withTimestamps in the source model.
6. Change the timestamp columns of the source table (with appropriate constant updates in the Model class)
6. Call the relation on Article and get() the results, this will result in an exception.

What was expected:
1. The relation would continue to work, without the pivot timestamp columns having to match.

Inside the repository I created a test that does step 7. I added a GitHub Action that runs the failing test: https://github.com/kassah/belongsToManyTimeStampIssue/runs/4296023048?check_suite_focus=true

I also set it to run the tests just prior to changing the source tables to demonstrate the test works just prior to updating the column names.
https://github.com/kassah/belongsToManyTimeStampIssue/runs/4296017471?check_suite_focus=true

**Comment by crynobone** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/172966/142986728-2ee8b6b7-f75d-4891-ac93-3a33e8a4fddb.png)

I believe the code is clear on this one, you should be using `withTimestamps('created_at', 'updated_at')`

**Reply by kassah** [reply]
This definitely resolves the immediate issue! thanks!
